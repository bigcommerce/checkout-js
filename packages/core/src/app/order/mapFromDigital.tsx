import { DigitalItem } from '@bigcommerce/checkout-sdk';
import React from 'react';

import { TranslatedString } from '../locale';

import getOrderSummaryItemImage from './getOrderSummaryItemImage';
import { OrderSummaryItemOption, OrderSummaryItemProps } from './OrderSummaryItem';

function mapFromDigital(item: DigitalItem): OrderSummaryItemProps {
    return {
        id: item.id,
        quantity: item.quantity,
        amount: item.extendedListPrice,
        amountAfterDiscount: item.extendedSalePrice,
        name: item.name,
        image: getOrderSummaryItemImage(item),
        productOptions: [
            ...(item.options || []).map(option => ({
                testId: 'cart-item-product-option',
                content: `${option.name} ${option.value}`,
            })),
            getDigitalItemDescription(item),
        ],
    };
}

function getDigitalItemDescription(item: DigitalItem): OrderSummaryItemOption {
    if (!item.downloadPageUrl) {
        return {
            testId: 'cart-item-digital-product',
            content: <TranslatedString id="cart.digital_item_text" />,
        };
    }

    return {
        testId: 'cart-item-digital-product-download',
        content:
            <a
                href={ item.downloadPageUrl }
                rel="noopener noreferrer"
                target="_blank"
            >
                <TranslatedString id="cart.downloads_action" />
            </a>,
    };
}

export default mapFromDigital;
