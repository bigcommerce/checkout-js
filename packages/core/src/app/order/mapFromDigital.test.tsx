import { LineItemOption } from '@bigcommerce/checkout-sdk';

import { getDigitalItem } from '../cart/lineItem.mock';

import mapFromDigital from './mapFromDigital';

describe('mapFromDigital()', () => {
    it('returns downloadable related properties when it has a download URL', () => {
        const { options, ...item } = getDigitalItem();
        const { productOptions = [] } = mapFromDigital(item);

        expect(productOptions[0].testId).toBe('cart-item-digital-product-download');

        expect(productOptions[0].content).toMatchSnapshot();
    });

    it('returns digital related properties when it has no download URL', () => {
        const item = {
            ...getDigitalItem(),
            options: [] as LineItemOption[],
            downloadPageUrl: '',
        };
        const { productOptions = [] } = mapFromDigital(item);

        expect(productOptions[0].testId).toBe('cart-item-digital-product');

        expect(productOptions[0].content).toMatchSnapshot();
    });
});
