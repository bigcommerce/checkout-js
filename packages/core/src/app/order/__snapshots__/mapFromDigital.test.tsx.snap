// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapFromDigital() returns digital related properties when it has no download URL 1`] = `
<TranslatedString
  id="cart.digital_item_text"
/>
`;

exports[`mapFromDigital() returns downloadable related properties when it has a download URL 1`] = `
<a
  href="url.php"
  rel="noopener noreferrer"
  target="_blank"
>
  <TranslatedString
    id="cart.downloads_action"
  />
</a>
`;
